src/main/java/io/github/acm19/aws/interceptor/http/RequestSigner.java

```diff
@@ -65,17 +64,18 @@ class RequestSigner {
      *
      * @param request to be signed
      * @return signed request
-     * @see Signer#sign
+     * @see AwsV4HttpSigner#sign
      */
-    SdkHttpFullRequest signRequest(SdkHttpFullRequest request) {
-        ExecutionAttributes attributes = new ExecutionAttributes();
-        attributes.putAttribute(AwsSignerExecutionAttribute.AWS_CREDENTIALS,
-                                awsCredentialsProvider.resolveCredentials());
-        attributes.putAttribute(AwsSignerExecutionAttribute.SERVICE_SIGNING_NAME, service);
-        attributes.putAttribute(AwsSignerExecutionAttribute.SIGNING_REGION, region);
+    SignedRequest signRequest(SdkHttpFullRequest request) {
+        SignedRequest signedRequest = signer.sign(r -> r.identity(awsCredentialsProvider.resolveCredentials())
+                .request(request)
+                .payload(request.contentStreamProvider().orElse(null))
+                .putProperty(AwsV4HttpSigner.SERVICE_SIGNING_NAME, service)
+                .putProperty(AwsV4HttpSigner.REGION_NAME, region.id())
+                .putProperty(AwsV4HttpSigner.DOUBLE_URL_ENCODE, false)
+                .putProperty(AwsV4HttpSigner.NORMALIZE_PATH, false));
```

Comment:
Why are you setting these two to false? Was there any type of request where the signature was failing?

Reply:
For s3 this these need to be false - i haven't actually tested with s3 (our use case is opensearch) but I thought I'd add it incase someone want's to use this for s3

Reply:
The lib is designed to support OpenSearch. I'm happy to have this added if some S3 user request it, but I feel like we shouldn't add S3 support unless there's some use case for it and then we should add test and documentation about it. So I'd leave these to their default value.

BTW, thanks for the contrib, it looks great to me. I'm happy to merge it once this and the other 2 outstanding comments by @dblock are addressed.

Reply:
+1 on leaving defaults and documenting (can be done in a separate PR, maybe open an issue if you're not going to do that one @GeorgeVincentDepop)

Btw, the two things I am aware of that both S3 and OpenSearch (Serverless) do differently is require `X-Amz-Content-Sha256`.